### PR TITLE
Allow JudgeTeams to be marked for new ACD priority job cases

### PR DESCRIPTION
**Pull request by lomky**
Connects #14604 

### Description
JudgeTeam organizations have a new boolean field to note whether they are available for automatically pushed priority cases.

### Acceptance Criteria
- [ ] Test suite passes
- [ ] JudgeTeams availability are set to true for existing & new judge teams
- [ ] when queried via scope, only the available JudgeTeams are returned

### Testing Plan
1. Run the migration
   - [ ] Confirm all existing JudgeTeams have `true` set for `automated_priority_case_distribution`
   - [ ] Confirm all other org types are `nil`
2. On Console, query `JudgeTeam.available_for_priority_case_distribution`
   - [ ] Confirm all JudgeTeams are returned
3. On Console, update one JudgeTeam to be unavailable `JudgeTeam.first.update(automated_priority_case_distribution: false)
4. On Console, query `JudgeTeam.available_for_priority_case_distribution`
   - [ ] Confirm all JudgeTeams less the unavailable one are returned

### Database Changes
*Only for Schema Changes*

* [x] Column comments updated
* [x] DB schema docs updated with `make docs` (after running `make migrate`)
* [x] #appeals-schema notified with summary and link to this PR

### CODE REVIEW

File: db/migrate/20200713194204_add_automated_priority_case_distribution_to_organization.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddAutomatedPriorityCaseDistributionToOrganization < ActiveRecord::Migration[5.2]
+  def change
+    add_column :organizations, :automated_priority_case_distribution, :boolean, comment: "Whether a JudgeTeam is currently available for automatically pushed priority cases"
```
Comment (yoomlam) [inline db/migrate/20200713194204_add_automated_priority_case_distribution_to_organization.rb:3]: My attempt to reduce words without losing meaning:
```suggestion
    add_column :organizations, :accepts_pushed_priority_cases, :boolean, comment: "Whether a JudgeTeam accepts automatically pushed priority cases"
```
We use [`priority_push`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/db/migrate/20200713173228_add_priority_to_distributions.rb#L3) to describe distributions, so I'm trying to use it here. An alternative is `:accepts_priority_pushed_cases` or use  `accepts priority_pushed automatic case distributions` in the comment to enable finding the field while grepping.

File: db/migrate/20200714185603_add_automated_priority_case_distribution_index_on_organization.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class AddAutomatedPriorityCaseDistributionIndexOnOrganization < ActiveRecord::Migration[5.2]
+  disable_ddl_transaction!
+
+  def change
+    add_index :organizations, :automated_priority_case_distribution, algorithm: :concurrently
```
Comment (yoomlam) [inline db/migrate/20200714185603_add_automated_priority_case_distribution_index_on_organization.rb:5]: TIL: https://thoughtbot.com/blog/how-to-create-postgres-indexes-concurrently-in
> creates the index without preventing concurrent INSERTs, UPDATEs, or DELETEs on the table ...

> [to] avoid accidental downtime caused by long, write-blocking database indexing

File: db/etl/migrate/20200714134734_add_automated_priority_case_distribution_to_etl_organization.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddAutomatedPriorityCaseDistributionToEtlOrganization < ActiveRecord::Migration[5.1]
+  def change
+    add_column :organizations, :automated_priority_case_distribution, :boolean, comment: "Whether a JudgeTeam is currently available for automatically pushed priority cases"
```
Comment (yoomlam) [inline db/etl/migrate/20200714134734_add_automated_priority_case_distribution_to_etl_organization.rb:3]: Reminder to update this file, if you change the original.

File: db/etl/migrate/20200714185603_add_automated_priority_case_distribution_index_on_organization.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class AddAutomatedPriorityCaseDistributionIndexOnOrganization < ActiveRecord::Migration[5.2]
+  disable_ddl_transaction!
+
+  def change
+    add_index :organizations, :automated_priority_case_distribution, algorithm: :concurrently
```
Comment (yoomlam) [inline db/etl/migrate/20200714185603_add_automated_priority_case_distribution_index_on_organization.rb:5]: Reminder to update this file, if you change the original.

File: app/models/organizations/judge_team.rb [hunk 5]
```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 class JudgeTeam < Organization
+  scope :available_for_priority_case_distribution, -> { where(automated_priority_case_distribution: true) }
```
Comment (yoomlam) [inline app/models/organizations/judge_team.rb:4]: Read this suggestion last.
```suggestion
  scope :pushed_priority_cases_allowed, -> { where(accepts_pushed_priority_cases: true) }
```

File: spec/models/organizations/judge_team_spec.rb [hunk 6]
```diff
@@ -136,6 +151,7 @@
           expect(judge.administered_teams.length).to eq(1)
           expect(judge.administered_teams.first.class).to eq(JudgeTeam)
           expect(judge.administered_teams.first.url).to eq(judge.css_id.downcase.parameterize.dasherize)
+          expect(judge.administered_teams.first.automated_priority_case_distribution).to eq(true)
```
Comment (yoomlam) [inline spec/models/organizations/judge_team_spec.rb:154]: May be a good candidate for a `shared_examples` with lines 31-35.

File: db/migrate/20200713194204_add_automated_priority_case_distribution_to_organization.rb (hunk 1 above)
Comment (lomky) [reply db/migrate/20200713194204_add_automated_priority_case_distribution_to_organization.rb:3]: :tada: I like :accepts_priority_pushed_cases for the grep synergy!

File: db/etl/schema.rb [hunk 7]
```diff
@@ -327,6 +327,7 @@
   end
 
   create_table "organizations", comment: "Copy of Organizations table", force: :cascade do |t|
+    t.boolean "accepts_priority_pushed_cases", comment: "Whether a JudgeTeam currently accepts distributiuon of automatically pushed priority cases"
```
Comment (yoomlam) [inline db/etl/schema.rb:330]: Typo:
```suggestion
    t.boolean "accepts_priority_pushed_cases", comment: "Whether a JudgeTeam currently accepts distribution of automatically pushed priority cases"
```

File: db/migrate/20200715105504_add_accepts_priority_pushed_cases_to_organization.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class AddAcceptsPriorityPushedCasesToOrganization < ActiveRecord::Migration[5.2]
+  def change
+    add_column :organizations, :accepts_priority_pushed_cases, :boolean, comment: "Whether a JudgeTeam currently accepts distributiuon of automatically pushed priority cases"
```
Comment (yoomlam) [inline db/migrate/20200715105504_add_accepts_priority_pushed_cases_to_organization.rb:3]:
```suggestion
    add_column :organizations, :accepts_priority_pushed_cases, :boolean, comment: "Whether a JudgeTeam currently accepts distribution of automatically pushed priority cases"
```

File: db/schema.rb [hunk 9]
```diff
@@ -890,7 +890,7 @@
   create_table "legacy_issue_optins", comment: "When a VACOLS issue from a legacy appeal is opted-in to AMA, this table keeps track of the related request_issue, and the status of processing the opt-in, or rollback if the request issue is removed from a Decision Review.", force: :cascade do |t|
     t.datetime "created_at", null: false, comment: "When a Request Issue is connected to a VACOLS issue on a legacy appeal, and the Veteran has agreed to withdraw their legacy appeals, a legacy_issue_optin is created at the time the Decision Review is successfully intaken. This is used to indicate that the legacy issue should subsequently be opted into AMA in VACOLS. "
     t.string "error"
-    t.date "folder_decision_date", comment: "Decision date on case record folder"
+    t.datetime "folder_decision_date", comment: "Date/Time of decision"
```
Comment (yoomlam) [inline db/schema.rb:893]: Exclude from this PR.
https://github.com/department-of-veterans-affairs/caseflow/pull/14652/files#r453719147

**Comment by yoomlam** [comment]
AC tests pass:
```ruby
Organization.where.not(type: :JudgeTeam).count
=> 0
Organization.where.not(type: :JudgeTeam).count
=> 20
Organization.where.not(type: :JudgeTeam).where(accepts_priority_pushed_cases: true).count
=> 0
Organization.where.not(type: :JudgeTeam).where(accepts_priority_pushed_cases: false).count
=> 0
JudgeTeam.count
=> 9
JudgeTeam.pushed_priority_cases_allowed.count
=> 9
JudgeTeam.third.update(accepts_priority_pushed_cases: false)
=> true
JudgeTeam.pushed_priority_cases_allowed.count
=> 8
JudgeTeam.pushed_priority_cases_allowed.include?(JudgeTeam.third)
=> false
```

File: spec/models/organizations/judge_team_spec.rb [hunk 10]
```diff
@@ -261,31 +229,14 @@
 
     context "a judge team with attorneys on it" do
       let(:user) { create(:user) }
-      let(:judge) { create(:user) }
-      let(:judge_team) { create(:judge_team) }
+      let(:judge) { Judge.new(user) }
```
Comment (yoomlam) [inline spec/models/organizations/judge_team_spec.rb:232]: Could condense 2 lines into one:
```ruby
let(:judge) { Judge.new(create(:user)) }
```

File: spec/models/organizations/judge_team_spec.rb [hunk 11]
```diff
@@ -261,31 +229,14 @@
...
-        it "returns the team judge" do
-          expect(judge_team.judge).to eq judge
-        end
-      end
-
-      describe ".attorneys" do
-        it "returns the team attorneys" do
-          expect(judge_team.attorneys).to match_array attorneys
-        end
+        populate_judgeteam_for_testing(judge_team, judge.user, attorneys)
```
Comment (yoomlam) [inline spec/models/organizations/judge_team_spec.rb:237]: Consistency:
```suggestion
        populate_judge_team_for_testing(judge_team, judge.user, attorneys)
```

File: db/etl/migrate/20200715110603_add_accepts_priority_pushed_cases_index_on_organization.rb [hunk 12]
```diff
@@ -0,0 +1,5 @@
+class AddAcceptsPriorityPushedCasesIndexOnOrganization < Caseflow::Migration
+  def change
+    add_safe_index :organizations, :accepts_priority_pushed_cases, algorithm: :concurrently
```
Comment (yoomlam) [inline db/etl/migrate/20200715110603_add_accepts_priority_pushed_cases_index_on_organization.rb:3]:
```suggestion
    add_safe_index :organizations, :accepts_priority_pushed_cases
```
You can leave `algorithm` out due to https://github.com/department-of-veterans-affairs/caseflow/blob/e171e539bbe5673e3bee679d377f5304d1d06302/lib/caseflow/migration.rb#L18

File: db/migrate/20200715105703_add_accepts_priority_pushed_cases_index_on_organization.rb [hunk 13]
```diff
@@ -0,0 +1,5 @@
+class AddAcceptsPriorityPushedCasesIndexOnOrganization < Caseflow::Migration
+  def change
+    add_safe_index :organizations, :accepts_priority_pushed_cases, algorithm: :concurrently
```
Comment (yoomlam) [inline db/migrate/20200715105703_add_accepts_priority_pushed_cases_index_on_organization.rb:3]:
```suggestion
    add_safe_index :organizations, :accepts_priority_pushed_cases
```

**Pull request merged**